qctrlopencontrols/dynamic_decoupling_sequences/predefined.py

```diff
@@ -679,13 +672,18 @@ def new_walsh_sequence(
     .. [#] `H. Ball and M. J Biercuk, EPJ Quantum Technol. 2, 11 (2015).
         <https://doi.org/10.1140/epjqt/s40507-015-0022-4>`_
     """
-    duration = _check_duration(duration)
-    paley_order = paley_order or 1
+
+    check_arguments(
+        duration > 0, "Sequence duration must be above zero.", {"duration": duration}
+    )
+    check_arguments(
+        1 <= paley_order < 2000,
```

Comment:
```suggestion
        1 <= paley_order <= 2000,
```
(to be consistent with the old check)